packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -0,0 +1,1459 @@
+        mapping(uint256 => uint128) kroShares;
+        mapping(uint256 => uint128) kghShares;
+        uint256[] undelegateRequestTimes;
+        mapping(uint256 => uint256[]) undelegateRequestedIds;
+        mapping(uint256 => uint128) undelegateRequests;
+    }
+
+    /**
+     * @notice Constructs the vault of a validator.
+     *
+     * @custom:field asset             Asset information of the vault.
+     * @custom:field reward            Reward information of the vault.
+     * @custom:field isInitiated       Whether the vault is initiated.
+     * @custom:field noSubmissionCount Number of counts that the validator did not submit the output in priority round.
+     * @custom:field kroDelegators     A mapping of validator address to KRO delegator struct.
+     * @custom:field kghDelegators     A mapping of validator address to KGH delegator struct.
+     */
+    struct Vault {
+        Asset asset;
+        Reward reward;
+        bool isInitiated;
+        uint8 noSubmissionCount;
+        mapping(address => KroDelegator) kroDelegators;
+        mapping(address => KghDelegator) kghDelegators;
+    }
+
+    /**
+     * @notice The numerator for the boosted reward.
+     */
+    uint128 public constant BOOSTED_REWARD_NUMERATOR = 40;
+
+    /**
+     * @notice The denominator for the boosted reward.
+     */
+    uint128 public constant BOOSTED_REWARD_DENOM = 100;
+
+    /**
+     * @notice The denominator for per-mille used at slashing and initialization of undelegation.
+     */
+    uint128 public constant PER_MILLE_DENOM = 1000;
+
+    /**
+     * @notice Virtual KRO amount per KGH.
+     */
+    uint128 public constant VKRO_PER_KGH = 100e18;
+
+    /**
+     * @notice The numerator of the tax.
+     */
+    uint128 public constant TAX_NUMERATOR = 20;
+
+    /**
+     * @notice The denominator of the tax.
+     */
+    uint128 public constant TAX_DENOMINATOR = 100;
+
+    /**
+     * @notice Decimals offset for the KRO and KGH shares.
+     */
+    uint128 public constant DECIMAL_OFFSET = 10 ** 6;
+
+    /**
+     * @notice The maximum number of KROs that can be undelegated at once.
+     */
+    uint256 public constant MAX_DELEGATE = 50_000_000 * 10 ** 18;
+
+    /**
+     * @notice Address of the KRO token contract.
+     */
+    IERC20 public immutable TOKEN;
+
+    /**
+     * @notice Address of the L2OutputOracle contract. Can be updated via upgrade.
+     */
+    L2OutputOracle public immutable L2_ORACLE;
+
+    /**
+     * @notice Address of the KGH token contract.
+     */
+    IERC721 public immutable KGH;
+
+    /**
+     * @notice Address of the KGHManager contract.
+     */
+    IKGHManager public immutable KGH_MANAGER;
+
+    /**
+     * @notice The address of the SecurityCouncil contract. Can be updated via upgrade.
+     */
+    address public immutable SECURITY_COUNCIL;
+
+    /**
+     * @notice The max number of reward recipients when distributing rewards.
+     */
+    uint128 public immutable MAX_REWARD_RECIPIENTS;
```

Comment:
This may seem like it means the number of delegates receiving rewards. How about renaming `MAX_OUTPUT_FINALIZATION`?

Reply:
That would be more intuitive, I'll fix it. Thanks!